source/security/minio-authentication-authorization-review.rst

```diff
@@ -0,0 +1,836 @@
+================================
+Authentication and Authorization
+================================
+
+.. default-domain:: minio
+
+.. contents:: On This Page
+   :local:
+   :depth: 2
+
+Overview
+--------
+
+*Authentication* is the process of verifying the identity of a connecting
+client. MinIO authentication requires providing user credentials in the form of
+an access key (username) and corresponding secret key (password). The MinIO
```

Comment:
Not really sure whether we should use the username / password analogy here. 
An S3 access/secret key pair is **explicitly** not a username /password - due to the S3 authentication method of computing the request authorization checksum (the value AWS S3 calls "signature"). S3 uses:
 ```
checksum := HMAC_SHA-256(serect_key, ...)
```
If an adversary observes one request checksum it can mount a (relatively cheap) brute-fore attack to recover the secret key - unless the secret key is chosen as random value of a sufficiently large domain: e.g. 256 bit random string.
HMAC-SHA256 is a function that is reasonably cheap / fast to compute and is not memory-hard. (It's a KDF, not a PBKDF).
Therefore, the access/secret key pair should be chosen at random -  not by a user as username/password.

Reply:
That's fair - I used the analogy because most users should be familiar with the general idea behind it (assuming they've never approached auth/authz before outside of normal user interactions w/ apps that require it). 

It also occurs to me that this implies we *only* support that kind of auth, where we do have a number of other methods (STS, JWT Web Identity, etc.) - though I guess those *also* boil down to auth + secret key?

Reply:
Yeah, so STS or `AssumeRole*` are basically mechanisms to exchange a e.g. STS token against a (temp). S3 access/secret key pair. The S3 protocol only knows about access/secret key pairs. Any other authentication method basically acts as adapter to connect systems that use different concepts (e.g. STS).

Reply:
Fixed